### PR TITLE
refactor: different config loading strategy

### PR DESCRIPTION
**Pull request by smashercosmo**
**Summary**

Fixes https://github.com/netlify/netlify-cms/issues/4817

In this refactoring config loading has been moved from App.js to bootstrap.js, which makes App component "dumber". And also redundant mergeConfig and configDidLoad actions has been removed. This also should make https://github.com/netlify/netlify-cms/pull/4776 merging easier.

**Test plan**

Run current tests

### CODE REVIEW

**Comment by smashercosmo** [comment]
Probably should add tests for loadConfig action

File: packages/netlify-cms-core/src/actions/config.js [hunk 1]
```diff
@@ -416,17 +406,16 @@ export async function handleLocalBackend(mergedConfig) {
...
   }
-  return async (dispatch, getState) => {
+  return async dispatch => {
     dispatch(configLoading());
 
     try {
-      const preloadedConfig = getState().config;
       const configUrl = getConfigUrl();
-      const isPreloaded = preloadedConfig && preloadedConfig.size > 1;
+      const isPreloaded = preloadedConfig && Object.keys(preloadedConfig).length > 1;
```
Comment (smashercosmo) [inline packages/netlify-cms-core/src/actions/config.js:418]: Should it be rather ```Object.keys(preloadedConfig).length > 0```?

**Comment by smashercosmo** [comment]
I believe this PR should also fix the problem with fast refresh. Currently when cms is being fast-refreshed, config validation results in error:
```
Unhandled Runtime Error
Error: 'collections[0].view_filters' should NOT have fewer than 1 items
'collections[0].view_groups' should NOT have fewer than 1 items
```
That's because after refresh, we validate existing config (which was normalized and to which defaults were applied), where each collection has view_filters and view_groups equal to empty array.

**Review by erezrokah — COMMENTED** [review]
Thanks again for this @smashercosmo, it's a tricky one, but the code looks much better now.

I added [another commit](https://github.com/netlify/netlify-cms/pull/4807/commits/313c91034e9d00e16b6fa66a4f7634fc3c4c0057) with some cleanup and a fix to [manual initialization](https://www.netlifycms.org/docs/beta-features/#manual-initialization).

Please let me know what you think.

**Comment by smashercosmo** [comment]
btw, did you have a chance to check if this fixes the issue with hot reload?

**Comment by erezrokah** [comment]
> btw, did you have a chance to check if this fixes the issue with hot reload?

Yes it does 🥳

**Review by erezrokah — APPROVED** [review]

**Pull request merged**